cores/rp2040/delay.cpp

```diff
@@ -21,30 +21,46 @@
 #include <pico.h>
 #include <pico/time.h>
 
-extern "C" void delay( unsigned long ms ) {
+#ifdef USE_TINYUSB
+#include "Adafruit_TinyUSB_API.h"
+#endif
+
+extern "C"
+{
+
+void yield(void) {
+    TinyUSB_Device_Task();
+    TinyUSB_Device_FlushCDC();
+}
+
+void delay( unsigned long ms ) {
     if (!ms) {
         return;
     }
 
-    sleep_ms(ms);
+    // Call wfe/wfi while have pending USB event can starve the task.
+    // If there is no alarm interrupt to help wake up MCU.
+    // TODO better implementation later
+    // More detail https://github.com/adafruit/circuitpython/pull/2956
+    while (ms--) {
+      yield();
```

Comment:
Note: since pico-sdk delay()/sleep() invoke wfi()/wfe(), extra effort is needed to ensure there is no pending usb events when needed similar to https://github.com/adafruit/circuitpython/pull/2956. Though I would like to have some feedback to see if this approach make sense to you first. Will fix it in later push.

Reply:
no longer needed since it run tud_task() in irq context now (see below comment for detail)